xl_auth/permission/views.py

```diff
@@ -61,6 +61,8 @@ def register(user_id, collection_id):
         else:
             flash_errors(register_permission_form)
 
+    if request.referrer and url_for('user.register') in request.referrer:
+        user_id = User.query.order_by(-User.id).all()[0].id
```

Comment:
Race condition. Two users going through this process at the same time can lead to unexpected results. Is there another way of figuring out which user was just created? Maybe something like "the latest user created by `current_user`"?

Reply:
Very clever!